### PR TITLE
Fixing Documentation issues 

### PR DESCRIPTION
**Pull request by thisiskaransgit**
**Notes for Reviewers**

This PR fixes 
- #7792 
- #7455

1. Indexed App import subcommand in mesheryctl reference
1. Corrected PowerShell command for setting environment variables
1. Fixed broken links 
1. Updated Guide Sidebar
1. Updated general contribution page 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 8057e5cdff213c2312b0787bccba13f9d1f656a5 at: https://64ac70134239d50a1bb440da--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit bff27ff7855cb17fcc46440116a1217b2b65e7e6 at: https://64ad621ee29dcf095a80cb9a--meshery-docs-preview.netlify.app

File: docs/_data/mesheryctlcommands/cmds.yml [hunk 1]
```diff
@@ -763,6 +763,18 @@ app:
...
+      description: Import the app manifest into Meshery
+      usage: |
+        mesheryctl app import [flags]
+      flags:
+        file:
+          name: --file, -f
+          description: Path/URL to app file
+        source-type:
+          name: --source-type, -s
+          description: source type of application
```
Comment (leecalcote) [inline docs/_data/mesheryctlcommands/cmds.yml:777]: @alphaX86 what's the plan for having this automated?

File: docs/_data/toc.yml [hunk 2]
```diff
@@ -119,12 +119,16 @@
   links:
     - title: Deploying Sample Applications
       url: guides/sample-apps
-    # - title: Running SMI Conformance Tests
-    #   url: guides/smi-conformance
+    - title: Running SMI Conformance Tests
+      url: guides/smi-conformance
```
Comment (leecalcote) [inline docs/_data/toc.yml:123]: Please remove this change, @thisiskaransgit

File: docs/_sass/code.scss [hunk 3]
```diff
@@ -71,7 +71,7 @@ pre {
     word-wrap: normal;
     background-color: var(--color-primary-medium);
     padding: 1rem;
-    overflow: scroll;
+    overflow: hidden;
```
Comment (leecalcote) [inline docs/_sass/code.scss:74]: How are you guaranteeing that content is being cutoff, @thisiskaransgit?

File: docs/pages/installation/quick-start.md [hunk 4]
```diff
@@ -66,7 +66,7 @@ Use the Meshery command line interface, [mesheryctl]({{ site.baseurl }}/guides/m
 You can visit the [Docker Hub](https://hub.docker.com/extensions/meshery/docker-extension-meshery) marketplace to directly install Meshery extension in your Docker Desktop.
 
 
-_Download, install, and run Meshery in a single command. See all [installation methods]({{ site.baseurl }}/installation/platforms)._
+_Download, install, and run Meshery in a single command. See all [installation methods]({{ site.baseurl }}/installation)._
```
Comment (leecalcote) [inline docs/pages/installation/quick-start.md:69]: Good.

File: docs/pages/project/contributing/contributing.md [hunk 5]
```diff
@@ -84,10 +96,18 @@ Practices for Production Environments](https://peter.bourgon.org/go-in-productio
...
+<ol>
+<li><code>Go</code> version 1.19 must be installed if you want to build and/or make changes to the existing code. The binary <code>go1.19</code> should be available in your path. If you don't want to disturb your existing version of Go, then follow these <a href="https://go.dev/doc manage-install#:~:text=and%20run%20them.-,Installing%20multiple%20Go%20versions,-You%20can%20install" rel="noopener" target="_blank">instructions</a> to keep multiple versions of Go in your system.</li>
+<li> <code>GOPATH</code> environment variable should be configured appropriately</li>
+<li> <code>npm</code> and <code>node</code> should be installed on your machine, `node` version 19 or higher is not supported right now.</li>
+<li>. Fork this repository <code>git clone https://github.com/meshery/meshery.git</code>, and clone your forked version of Meshery to your development environment, preferably outside `GOPATH`.</li>
+<li> <code>golangci-lint</code> should be installed if you want to test Go code, for MacOS and linux users.</li>
+<li> <code>golangci-lint</code> should be installed if you want to test Go code, for MacOS and linux users.</li>
+</ol>
+
+</details>
```
Comment (leecalcote) [inline docs/pages/project/contributing/contributing.md:113]: Good

File: docs/pages/reference/mesheryctl-commands.md [hunk 6]
```diff
@@ -633,6 +633,19 @@ Installation, troubleshooting and debugging of Meshery and its adapters.
...
+        <td></td>
+        <td>{{ subcommand5.description }}</td>
+      </tr>
+      {% for flag_hash in subcommand5.flags %}{% assign flag = flag_hash[1] %}
+        <tr>
+          <td></td>
+          <td>{{ flag.name }}</td>
+          <td>{{ flag.description }}</td>
+      </tr>
+      {% endfor %}
```
Comment (leecalcote) [inline docs/pages/reference/mesheryctl-commands.md:648]: @alphaX86 please review.

File: docs/pages/guides/configuration-management.md [hunk 7]
```diff
@@ -55,7 +55,7 @@ mesheryctl pattern apply BookInfoApp
 
 This will apply the pattern BookInfoApp, which has already been imported into Meshery.
 
-See [mesheryctl Command Reference](../reference/mesheryctl/subcommands/mesheryctl-pattern-apply.md) for more details on the `pattern` subcommand.
+See [mesheryctl Command Reference](../reference/mesheryctl/#cloud-native-pattern-configuration-and-management) for more details on the `pattern` subcommand.
```
Comment (leecalcote) [inline docs/pages/guides/configuration-management.md:58]: @alphaX86 please review.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: docs/_sass/code.scss (hunk 3 above)
Comment (thisiskaransgit) [reply docs/_sass/code.scss:74]: | Before   |      After      |
|----------|:-------------:|
|  |   |
This won't have side effects in other components

**Comment by l5io** [comment]
🚀 Preview for commit 0ddfd489ad0c4bf5a789cf58fab5cf6629fbff5d at: https://64aec62f56374303439bce82--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 6818d07dd89bd9b384b88e0f2b44230a50f1b4a2 at: https://64aecb4dc5feb014139275c8--meshery-docs-preview.netlify.app

**Comment by vishalvivekm** [comment]
@thisiskaransgit Let's discuss it on the websites call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima), if you would :)

**Comment by l5io** [comment]
🚀 Preview for commit a4c416b1551d6f96f7bcec6f20fd5a28c2452365 at: https://64b85db3fac8c80396ab4025--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 5f02a06403dbc37d80645e531a246f5fd1927265 at: https://64b94f9114192800a1eca30b--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit db3192b674a33139fcfca5424cab8cd32e9fb518 at: https://64b9a3893beda942e4e7c983--meshery-docs-preview.netlify.app

File: docs/_data/mesheryctlcommands/cmds.yml (hunk 1 above)
Comment (alphaX86) [reply docs/_data/mesheryctlcommands/cmds.yml:777]: I have no qualms on making this automated but we would lose the desirability of the format we're having if we make this automated.

Through this tree format only, the index is being updated properly (even though it's manual process). If we want to make this automated, we want to either change the structure of index page (both YML and HTML page) or making a use of 3rd party tool. Maybe LLMs have the answer to this problem

File: docs/_sass/code.scss (hunk 3 above)
Comment (alphaX86) [reply docs/_sass/code.scss:86]: This seems good

File: docs/pages/guides/configuration-management.md (hunk 7 above)
Comment (alphaX86) [reply docs/pages/guides/configuration-management.md:58]: Desired one. Still....

```suggestion
See [mesheryctl pattern subcommand section](../reference/mesheryctl/#cloud-native-pattern-configuration-and-management) for more details on the `pattern` subcommand.
```

File: docs/pages/reference/mesheryctl-commands.md (hunk 6 above)
Comment (alphaX86) [reply docs/pages/reference/mesheryctl-commands.md:648]: Great job. 

We would still need to find a way for automating this though to avoid manual process

**Review by alphaX86 — COMMENTED** [review]
Got to review today coz of WFH. Here's my opinions

**Comment by l5io** [comment]
🚀 Preview for commit d831489cd5828c5feab8e162b36b5ff6a78d4225 at: https://64ba15f8c60d8e3ca9dc5c18--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 520ffc8e1f2acb1abc5e77930b02982f016fdc27 at: https://64bed74e6a0ca814dec7a55d--meshery-docs-preview.netlify.app

**Comment by thisiskaransgit** [comment]
@leecalcote, looks like we're good to go with this one

**Review by leecalcote — APPROVED** [review]
Thank you, @thisiskaransgit

**Review by alphaX86 — APPROVED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit dccdea46887f0617ab9a1b1038f84b382c4dc615 at: https://64e0e8bd011d58612e0f13f4--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
Thanks @thisiskaransgit

**Pull request merged**